### ISSUE TITLE
[Proposal] Add a new method for retrieving dependency modules

### ISSUE THREAD

**Issue by dos65** [build-server-protocol/build-server-protocol#164]
### Motivation

Currently, BSP clients have a way to get dependencies and source artifacts: `ScalaTarget.classpath` / `DependencySourcesItem.sources`.
However, these fields have`Array[Uri/String]` type and there is no way to get a meta-information from them like `organization`, `artifact-name`, `version`.

For example, in the case of having this information, Metals might represent dependency sources in a more convenient way for users. Instead of extracting  them directly into `.readonly` it might add an additional root directory :`.readonly/$org:$artifact$version`. Also, this thing would help internally for indexing purposes.

### Proposal

Add an additional optional method to BSP - `buildTarget/dependencyModules` that returns a list of dependency modules with attached meta information. 

The sample structure might be the following one (I've copied it from bloop configuration format):
```ts
export interface DependencyModulesResult {
  items: DependencyModuleItem[];
}

export interface DependencyModuleItem {
  target: BuildTargetIdentifier;
  moduels: DependencyModule[];
}

export interface DependencyModule {
   organization: String;
   name: String;
   version: String;
   artifacts: Artifact[] ;
}

export interface Artifact {
   path: Uri;
   // `sources`/`docs` or empty
   classifier?: String;
}

```

**Comment by jastice** [comment]
I think this would be a reasonable addition to the JVM extension

**Comment by jvican** [comment]
Fixed in #165
